### ISSUE TITLE
Listening To Radio Using OSD - Dash Always Appears

### ISSUE THREAD

**Issue by 0pLuS0** [DeaDBeeF-Player/deadbeef#2991]
### Steps to reproduce the problem

I'm listening to 80s Forever Radio https://www.80sforever.radio/how-to-listen/#fwdurpPlayer0?trackid=0 and added in this URL;

```http://azuracast.ssl-stream.com:8050/listen.pls?sid=1```

I also tried this URL; ```http://azuracast.ssl-stream.com:8050/listen.m3u?sid=1```

Then open deadbeef and play and enable the OSD.

### What's going on? Describe the problem in as much detail as possible.

The OSD always displays a dash ```-```  after the Station and Artist name.

In this screen shot, when I first start deadbeef and click on play, the Station name appears and then below it a dash.

![station_osd](https://github.com/DeaDBeeF-Player/deadbeef/assets/36190223/39a70661-0fdc-4aa4-9997-7be87e732ed1)

Here's the OSD displaying a song title and the artist named Polecats, with a dash after it.

![Radio OSD](https://github.com/DeaDBeeF-Player/deadbeef/assets/36190223/fc09f144-ed16-4649-bd13-c4d81d325365)

### Information about the software:

Deadbeef version: 1.9.5
OS: Slackware 15.0

### Expected outcome

No dash appears after the station's or artist's name.

THANKS

**Comment by 0pLuS0** [comment]
@Oleksiy-Yakovenko 

Hi, Something else I noticed, sometimes when I start deadbeef the OSD will display for the radio station and songs two times, instead of only one popup for each.

**Comment by Npa972** [comment]
Same for me (two time display)
DeadBeeF-devel, archlinux

**Comment by 0pLuS0** [comment]
Hi @Oleksiy-Yakovenko if 1.9.7 is the next version, should these issues be corrected then?

THANKS

**Comment by Oleksiy-Yakovenko** [comment]
I did not look into this issue yet. So most likely it is not fixed yet.

**Comment by 0pLuS0** [comment]
> I did not look into this issue yet. So most likely it is not fixed yet.

THANKS for all your hardwork, it's greatly appreciated!

I love deadbeef! 👍

**Comment by 0pLuS0** [comment]
Hi @Oleksiy-Yakovenko 

Can I just drop the notify.c into 1.9.6 and compile it?

I tried this, but I still get the dashes...

THANKS

**Comment by 0pLuS0** [comment]
Hi @Oleksiy-Yakovenko 

I don't know if you saw my reply last year? If this is the fix for notify.c 

https://github.com/DeaDBeeF-Player/deadbeef/commit/d3111ac795fba3da465dac9b9f002115ba931ee5

I tried to add that into 1.9.6 but I wasn't able to compile it.

Should I be able to use that fix in 1.9.6 and compile it?

THANKS

**Comment by Oleksiy-Yakovenko** [comment]
I didn't put any effort to support such scenario - so no, it's not intended to work. If it works - it's by pure accident.

**Comment by 0pLuS0** [comment]
Hi @Oleksiy-Yakovenko 

I thought this was the fix for the next update?

https://github.com/DeaDBeeF-Player/deadbeef/commit/d3111ac795fba3da465dac9b9f002115ba931ee5

**Comment by Oleksiy-Yakovenko** [comment]
My last reply is about whether it's supposed to work in 1.9.6, to which I answered no.

It's indeed supposed to work in new builds / future release.

**Comment by 0pLuS0** [comment]
Hello @Oleksiy-Yakovenko 

I just compiled 1.10.0 and it still shows the dash...

Was the fix suppose to be in this version?

THANKS

**Comment by 0pLuS0** [comment]
Hi @Oleksiy-Yakovenko 

I went and had a look again at the OSD Notify Plugin options and noticed the ``` Notification content format %artist% - %album%``` has a dash in the middle.

![Image](https://github.com/user-attachments/assets/8884768d-5e59-4cf7-a4ab-beb541540cf0)

I have not changed these settings, they are the default, but if I make Notification content format without the dash as;

```%artist% %album%```

I am not seeing this issue...

![Image](https://github.com/user-attachments/assets/4827f4bc-7332-4202-a9de-94b450dec111)

What is the correct format?

```%artist% %album%``` or ```%artist% - %album%```

THANKS

**Comment by Oleksiy-Yakovenko** [comment]
I think you need to click the reset to defaults button to get the new default title format.

**Comment by 0pLuS0** [comment]
Hi @Oleksiy-Yakovenko 

I clicked Reset but it still shows the format as; ```%artist% - %album%```

![Image](https://github.com/user-attachments/assets/e9928b67-0628-4451-86d6-f3992068bd06)


I still see the Dash - appear after the artist name using the default format.


![Image](https://github.com/user-attachments/assets/89b314d6-546e-4583-a40c-c652b23954c7)

**Comment by Oleksiy-Yakovenko** [comment]
ok.. That's odd -- I'll double check.
Regardless, the new title format that fixes dash is this:
```%artist%$if($and(%artist%,%album%), - ,)%album%```

**Comment by 0pLuS0** [comment]
Hi @Oleksiy-Yakovenko 

Is it ok for me to just add it to the settings. or it needs to be compiled from source, built into it?

For now I put it in...

THANKS

![Image](https://github.com/user-attachments/assets/07ff2ba2-3783-4095-9dad-0449637c9102)

**Comment by Oleksiy-Yakovenko** [comment]
Yes, it should work if you put it there manually. I'll check why reset doesn't work.

**Comment by Oleksiy-Yakovenko** [comment]
I believe this is fixed? Closing for now.
